src/descriptor/mod.rs

```diff
@@ -792,6 +792,42 @@ impl Descriptor<DescriptorPublicKey> {
     }
 }
 
+impl<Pk: MiniscriptKey> Descriptor<Pk> {
+    /// Whether this descriptor is a multipath descriptor that contains any 2 multipath keys
+    /// with a different number of derivation paths.
+    /// Such a descriptor is invalid according to BIP389.
+    pub fn multipath_length_mismatch(&self) -> bool {
```

Comment:
Just a review comment. Nothing to be addressed here. Ideally, we want to make sure that such invalid descriptors cannot be created in the first place. Adding this pub might confuse users to call it to double-check everything. 

I think this is true for all APIs that deal with Miniscript, but not for the internal `Terminal`. I have some changes in mind in future PRs to remove all the confusing sanity_check/check APIs.